test/main.lua

```diff
@@ -354,6 +461,55 @@ function main()
     print(('%d test files failed to load'):format(counts.file_errors))
 end
 
-script.start(function()
-    dfhack.with_finalize(finish_tests, main)
-end)
+local function main(args)
+    local help, nocache, test_dir, mode_filter, test_filter =
+            false, false, nil, {}, {}
+    local other_args = utils.processArgsGetopt(args, {
+            {'h', 'help', handler=function() help = true end},
+            {'d', 'test_dir', hasArg=true,
+            handler=function(arg) test_dir = arg end},
+            {'n', 'nocache', handler=function() nocache = true end},
+            {'m', 'modes', hasArg=true,
+            handler=function(arg) mode_filter = arg:split(',') end},
+            {'t', 'tests', hasArg=true,
+            handler=function(arg) test_filter = arg:split(',') end},
+        })
+
+    if help then print(help_text) return end
+
+    local done_command = table.concat(other_args, ' ')
+    local config = load_test_config(CONFIG_FILE)
+
+    -- override config with any params specified on the commandline
+    if test_dir then config.test_dir = test_dir end
+    if nocache then config.nocache = true end
```

Comment:
Honestly, I might end up inverting this. CI always runs tests through `run-tests.py`, which could pass a `--cache` flag to `test/main.lua` fairly easily and wouldn't need CI changes.

Reply:
that works. maybe rename `-n, --nocache` to `-r, --resume`. Want me to do that?

Reply:
Makes sense to me. Feel free